addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -3336,7 +3336,9 @@ export class OdooEditor extends EventTarget {
                         const textFragments = splitAroundUrl[i].split(/\r?\n/);
                         let textIndex = 1;
                         for (const textFragment of textFragments) {
-                            this._applyCommand('insertText', textFragment);
+                            // Replaces consecutive spaces in a text fragment with non-breaking spaces.
+                            const modifiedTextFragment = textFragment.replace(/ {2,}/g, match => '\u00A0'.repeat(match.length));
```

Comment:
You can't replace all spaces by non-breaking ones because that would mean the system can never break the line, then. What we usually do is alternating between nbsp and normal space so that the system can break on the normal ones if needed. See test results below.